### ISSUE TITLE
Snap Camera does not show up as "device" in apps ie discord, OBS

### ISSUE THREAD

**Issue by RaincloudLabs** [jaku/SnapCameraPreservation#54]
Hi!
App is working great and filters show up, within snap camera app.  When attempting to use the virtual camera in Discord or OBS it does not show up as a device.
Ideas?
Thanks!

**Comment by dpkonofa** [comment]
This is probably a code-signing issue and not something caused by SCP unless it was working before. Are you on a Mac or a PC?

**Comment by RaincloudLabs** [comment]
Mac M2 Pro running Ventura

**Comment by dpkonofa** [comment]
Definitely a code signing issue. Probably related to this: https://stackoverflow.com/questions/61114738/are-macos-virtual-webcams-inherently-incompatible-with-10-14s-hardened-runtime

**Comment by RaincloudLabs** [comment]
its killing me lol

**Comment by RaincloudLabs** [comment]
I've removed all permissions
 % sudo codesign --remove-signature "/Applications/Discord.app/Contents/Frameworks/Discord Helper (Renderer).app"
% sudo codesign --remove-signature "/Applications/Discord.app/Contents/Frameworks/Discord Helper (GPU).app"
% sudo codesign --remove-signature "/Applications/Discord.app/Contents/Frameworks/Discord Helper (Plugin).app" 
and still nothing

**Comment by mololami** [comment]
I'm having the same issue since last week. It has previously been working for my M1 macbook air on ventura, now it won't register as a device on any platform. Did you figure out a solution?

**Comment by mololami** [comment]
@jaku would you have any ideas maybe? I've restarted my laptop so many times, deleted snapcam and all existing files several times too thinking the next time will fix it, and it just won't

**Comment by jaku** [comment]
What's seemed to have worked for users in the past is going into privacy settings under System Preferences, remove all camera permissions on the apps you're trying to use (Discord).

Reinstall Snap Camera, you don't need to refollow the steps to get lenses working again. And then the next time you try to use the Camera in Discord it should prompt you to allow permissions and it should work.

**Comment by RaincloudLabs** [comment]
Thank you for the idea.  The main issue is that the snap camera does not show up as an option in other apps like Discord of OBS.  All other streaming softwares I have tried have shown up except SC.  
![Screenshot 2023-05-23 at 11 16 49 PM](https://github.com/jaku/SnapCameraPreservation/assets/15900999/85efed2a-da71-4b55-a7f0-139c49e9b3fa)

**Comment by mololami** [comment]
Me too @RaincloudLabs. Snapcam is the only virtual camera that doesn't show up, out of all of mine despite all permissions being on.

**Comment by d0x7** [comment]
While it does work for me in SnapCam and I can select the camera in OBS, it's not showing anything. It's black once I select it and then after some time, maybe 20 seconds it changes to this blank yellow screen (not even the normal placeholder image when SnapCam is not running that I know back from windows).

Any ideas?

<img width="1082" alt="image" src="https://github.com/jaku/SnapCameraPreservation/assets/17951467/16306951-bd55-4e48-98ff-4434de1b9141">

**Comment by RaincloudLabs** [comment]
Did you do something to make it show up in OBS?  
Did you simply just install and add the server cert? 
Thanks

**Comment by d0x7** [comment]
Yeah basically, it's been a while since I installed it (wanted to try again yesterday but still not working), but I just followed the README, installed the cert and it showed up like normal within OBS, although I can select it neither in Discord nor Photo Booth, so there are definitely issues with it.

**Comment by RaincloudLabs** [comment]
Working!!! d0x7
Nuked my entire computer today cause it was sluggish.  Set up as new.  Redownloaded everything. 
it wasn't working in OBS but I changed the resolution to match that of snap camera (1280x720) and boom works great.  Unsigned and resigned Discord and it is now show OBS Cam and Snap is working great!!!

**Comment by mololami** [comment]
Thats awesome @RaincloudLabs !!! Can I ask what you mean by "nuked my whole computer"? And by unsign resign Discord do you mean literally just signing out or deleting the app? I've had the same issue as you and I would like to try exactly what you've done.

Also - is 720p the highest resolution you can go with snapcam?

**Comment by d0x7** [comment]
> Working!!! 

Very interessting, but I sadly cannot get it to work. I created a Video Capture Device in OBS, selected Snap and I tried both a preset with 720p and not using a Preset at all at 720p, but still the yellow screen with either. Enabling/disabling buffering didn't change anything. I might try reinstalling both OBS/Snap, but cant/dont wanna reinstall MacOS right now. Did you do anything different from what I tried/wrote?

**Comment by NewboyAndCat** [comment]
I'm going to put an end to that. I've been agonizing over it all night. But I've fixed it now, and it works fine, no more yellow screen issues. The way I did it was I backed up lens and then I cleared snap's cache. Then completely delete the snap camera, re-follow the tutorial to install the replacement signature and other actions, the same process, and then he can run, now very smooth. Hope to help all the friends who are troubled by this problem
